optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/spotroster/SpotRosterViewportFactory.java

```diff
@@ -149,12 +149,80 @@ private ShiftBlob buildShiftBlob(final Spot spot,
                                      final TimeSlot timeSlot,
                                      final Employee employee) {
 
-        final Shift shift = new Shift(shiftView, spot, timeSlot);
+        final Shift shift = new Shift(shiftView, spot, timeSlot.getStartDateTime(), timeSlot.getEndDateTime());
         shift.setEmployee(employee);
         return new ShiftBlob(scale, shift);
     }
 
     private <T extends AbstractPersistable> Map<Long, T> indexById(final List<T> abstractPersistables) {
         return abstractPersistables.stream().collect(toMap(AbstractPersistable::getId, identity()));
     }
+    
+    private static final class TimeSlot {
```

Comment:
I think this class should exist anymore.. the "spotRosterModel" becomes a `Map<Spot, List<ShiftView>>`

Reply:
Will update

Reply:
done